lib/AllocateDescriptorsPass.cpp

```diff
@@ -603,11 +603,10 @@ bool clspv::AllocateDescriptorsPass::AllocateKernelArgDescriptors(Module &M) {
 
         if (!var_fn) {
           // Make the function
-          // TODO: #816 remove after final transition.
           Type *ret_ty = nullptr;
           if (Arg.getType()->isTargetExtTy()) {
             ret_ty = Arg.getType();
-          } else if (Arg.getType()->isOpaquePointerTy()) {
+          } else if (Arg.getType()->isPointerTy()) {
             ret_ty = PointerType::get(M.getContext(), addr_space);
           } else {
             ret_ty = resource_type->getPointerTo(addr_space);
```

Comment:
This should really be:
```suggestion
          } else if (Arg.getType()->isPointerTy()) {
            ret_ty = PointerType::get(M.getContext(), addr_space);
```